hackathon_site/dashboard/frontend/src/slices/event/teamDetailSlice.ts

```diff
@@ -30,6 +32,44 @@ interface RejectValue {
     message: any;
 }
 
+interface idProvidedParameters {
+    profileId: number;
+    idProvided: boolean;
+}
+
+export const updateParticipantIdProvided = createAsyncThunk<
+    Profile,
+    idProvidedParameters,
+    { state: RootState; rejectValue: RejectValue; dispatch: AppDispatch }
+>(
+    `${teamDetailReducerName}/updateParticipantIdProvided`,
+    async ({ profileId, idProvided }, { rejectWithValue, dispatch }) => {
+        try {
+            const response = await patch<Profile>(`/api/event/profiles/${profileId}/`, {
+                id_provided: idProvided,
+            });
+            return response.data;
+        } catch (e: any) {
```

Comment:
If the profile not found, the error msg will show up correctly, but the loading state hasn't been updated. So the table stays in "disabled" state.

Reply:
If we encountered the "Not Found" error, it means there is a profile updated in the database. We would need to get the most up-to-date profile from API. I would suggest querying for the profiles again here. When the most up-to-date profiles from API are retrieved, release the loading state.

Reply:
you're right about the loading state. i added in a rejected case to set loading to false.

not sure i understand your second comment though @Leo6Leo. if we encounter "Not Found", that means there is no profile in the database matching the id that we queried for. in this case, i don't think anything gets updated in the database, because there is no matching profile to update. could you clarify what you mean and what you'd like me to change?

Reply:
When we encounter "Not Found”, it means the information on the current webpage is not most up-to-date. We need to refresh the table to see the most up-to-date information in the database. Does this clear?

Reply:
@Leo6Leo not found just means the team doesn't exist like if you go to `/teams/12345` and team 12345 doesn't exist then the Not Fond error will appear

Reply:
The reason why I mentioned this is because I did the following things:
1. open Luke's table, and the profiles displayed
2. Delete the profile in the database
At this time, the profile will still be there on Luke's table because we didn't refresh the page. When we click on the deleted profile's checkbox, it will say "Not Found". 

What I would suggest is to refresh the table after the 'Not Found" error appears.

Reply:
ah okay makes sense 👍

Reply:
yes i understand now... i'll try just dispatching the thunk again inside the catch block if i get the "Not Found" error message

Reply:
sounds good Luke

Reply:
or maybe try to call `getTeamInfoData` everytime when we have executed the `updateParticipantIdProvided` thunk? So that the table will always remain most up-to-date.

Reply:
this could also solve the problem like this:

1. open Luke's table, and the profiles displayed
2. Add one more new team member
At this time, the new team member's profile will not be there on Luke's table because we didn't refresh the page.

Reply:
yes that sounds good.

the only problem is that `updateParticipantIdProvided` gets called in the `teamInfoTable` component, but that component doesn't have access to the teamCode, so i can't call the `getTeamInfoData` thunk. i could pass teamCode down from the `TeamDetail` as props, but idk if this is good practice or not. are you ok with me doing this leo?

Reply:
eh nvm imma just do it 😃

Reply:
@Leo6Leo so i tried calling both thunks every time a checkbox is clicked, but it doesn't work very well because the table reloads (loading bar) every time i click the checkbox. even if the users in the table don't change, it still triggers the loading bar because by calling the `getTeamInfoData` thunk, i'm updating the loading state of the table to true, so it displays the loading bar even though it doesn't have to. i think it's simpler to do it how i had it before, and admins will just have to refresh the page if a new user joins the team

Reply:
sounds good. I agree with u and Yuying that we should leave it as is. And the thing I mentioned could be made as a ticket for the future.